### PR TITLE
Make code test coverage visible to sonarCloud 

### PR DESCRIPTION
**Pull request by jfonseca-aneo**
Fix https://github.com/aneoconsulting/ArmoniK/issues/122

- The coverage report is generated with [dotnet-coverage](https://docs.microsoft.com/en-us/dotnet/core/additional-tools/dotnet-coverage) and passed to sonarCloud.
- Locally, one can generate a html version of the report with [dotnet-reportgenerator](https://www.nuget.org/packages/dotnet-reportgenerator-globaltool) for a nicer visualization.

### CODE REVIEW

File: dotnet-coverage.runsettings.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Configuration>
+    <CodeCoverage>
+        <ModulePaths>
+            <Include>
+                <ModulePath>.*\.dll$</ModulePath>
+                <ModulePath>.*\.exe$</ModulePath>
+            </Include>
+	        <Exclude>
+		        <ModulePath>.*Moq.dll$</ModulePath>
```
Comment (lemaitre-aneo) [inline dotnet-coverage.runsettings.xml:10]: indentation mismatch

**Review by lemaitre-aneo — COMMENTED** [review]
What is the prupose of `"System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute"`? And why is it added to all projects?

File: Compute/PollingAgent/src/ArmoniK.Core.Compute.PollingAgent.csproj [hunk 2]
```diff
@@ -49,4 +49,7 @@
     <ProjectReference Include="..\..\..\Common\src\ArmoniK.Core.Common.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+	<AssemblyAttribute Include="System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute" />
```
Comment (lemaitre-aneo) [inline Compute/PollingAgent/src/ArmoniK.Core.Compute.PollingAgent.csproj:53]: Why is this (non-test) project is excluded from coverage?

File: Control/Metrics/src/ArmoniK.Core.Control.Metrics.csproj [hunk 3]
```diff
@@ -44,5 +44,8 @@
     <ProjectReference Include="..\..\..\Adaptors\Redis\src\ArmoniK.Core.Adapters.Redis.csproj" />
     <ProjectReference Include="..\..\..\Common\src\ArmoniK.Core.Common.csproj" />
   </ItemGroup>
-
+	
+  <ItemGroup>
+	<AssemblyAttribute Include="System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute" />
```
Comment (lemaitre-aneo) [inline Control/Metrics/src/ArmoniK.Core.Control.Metrics.csproj:49]: Why is this (non-test) project is excluded from coverage?

File: Control/Submitter/src/ArmoniK.Core.Control.Submitter.csproj [hunk 4]
```diff
@@ -48,4 +48,8 @@
     <ProjectReference Include="..\..\..\Common\src\ArmoniK.Core.Common.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+	<AssemblyAttribute Include="System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute" />
```
Comment (lemaitre-aneo) [inline Control/Submitter/src/ArmoniK.Core.Control.Submitter.csproj:52]: Why is this (non-test) project is excluded from coverage?

File: ProfilingTools/src/ArmoniK.Core.ProfilingTools.csproj [hunk 5]
```diff
@@ -36,4 +36,8 @@
     <ProjectReference Include="..\..\Adaptors\MongoDB\src\ArmoniK.Core.Adapters.MongoDB.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+	<AssemblyAttribute Include="System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute" />
```
Comment (lemaitre-aneo) [inline ProfilingTools/src/ArmoniK.Core.ProfilingTools.csproj:40]: Why is this (non-test) project is excluded from coverage?

**Review by lemaitre-aneo — COMMENTED** [review]
Some non-tests projects seem to excluded from coverage

File: Compute/PollingAgent/src/ArmoniK.Core.Compute.PollingAgent.csproj (hunk 2 above)
Comment (jfonseca-aneo) [reply Compute/PollingAgent/src/ArmoniK.Core.Compute.PollingAgent.csproj:53]: This project contains the entry point for a polling agent, main methods should not be tested but the methods it calls. Hence, I think they should be excluded from coverage.

File: Control/Submitter/src/ArmoniK.Core.Control.Submitter.csproj (hunk 4 above)
Comment (jfonseca-aneo) [reply Control/Submitter/src/ArmoniK.Core.Control.Submitter.csproj:52]: This project contains the entry point for a submitter, main methods should not be tested but the methods it calls. Hence, I think they should be excluded from coverage.

File: Control/Metrics/src/ArmoniK.Core.Control.Metrics.csproj (hunk 3 above)
Comment (jfonseca-aneo) [reply Control/Metrics/src/ArmoniK.Core.Control.Metrics.csproj:49]: This is a helper project to collect metrics, I don't think its part of the core functionalities of the library so it should not be tested by us.

File: ProfilingTools/src/ArmoniK.Core.ProfilingTools.csproj (hunk 5 above)
Comment (jfonseca-aneo) [reply ProfilingTools/src/ArmoniK.Core.ProfilingTools.csproj:40]: This is also a helper project to collect traces, I don't think its part of the core functionalities of the library so it should not be tested.

**Pull request merged**